src/iuse_actor.cpp

```diff
@@ -2578,6 +2609,17 @@ repair_item_actor::attempt_hint repair_item_actor::repair( player &pl, item &too
         roll = NEUTRAL;
     }
 
+    if (t->has_quality(quality_sew)) {
+        p->add_msg_if_player(m_info,
+            _("You have used %d thread charges."), roll == SUCCESS ? filament_required : 1);
+        for (const auto &it : roll == SUCCESS ? req_success.get_components() : req.get_components() ) {
+            p->consume_items(it);
```

Comment:
Game asks for components each repair tick now - I know that's very bad and I will change that.